### ISSUE TITLE
Fail to build for powerpc linux

### ISSUE THREAD

**Issue by marillat** [gpac/gpac#2363]
Debian unstable powerpc
GCC 12.2.0

If you have an idea, because I don't know what is wrong.
````
evg/stencil.c: In function ‘get_pix_yuyv_10’:
evg/stencil.c:1863:43: error: subscripted value is neither array nor pointer nor vector
 1863 | #define GET_LE_10BIT_LEFT_AS_8(_ptr) (_ptr[1] )
      |                                           ^
evg/stencil.c:2220:14: note: in expansion of macro ‘GET_LE_10BIT_LEFT_AS_8’
 2220 |         vu = GET_LE_10BIT_LEFT_AS_8(p_src + _this->off_u);
      |              ^~~~~~~~~~~~~~~~~~~~~~
evg/stencil.c:1863:43: error: subscripted value is neither array nor pointer nor vector
 1863 | #define GET_LE_10BIT_LEFT_AS_8(_ptr) (_ptr[1] )
      |                                           ^
evg/stencil.c:2222:22: note: in expansion of macro ‘GET_LE_10BIT_LEFT_AS_8’
 2222 |                 vy = GET_LE_10BIT_LEFT_AS_8(p_src + _this->off_y+4);
      |                      ^~~~~~~~~~~~~~~~~~~~~~
evg/stencil.c:1863:43: error: subscripted value is neither array nor pointer nor vector
 1863 | #define GET_LE_10BIT_LEFT_AS_8(_ptr) (_ptr[1] )
      |                                           ^
evg/stencil.c:2224:22: note: in expansion of macro ‘GET_LE_10BIT_LEFT_AS_8’
 2224 |                 vy = GET_LE_10BIT_LEFT_AS_8(p_src + _this->off_y);
      |                      ^~~~~~~~~~~~~~~~~~~~~~
evg/stencil.c:1863:43: error: subscripted value is neither array nor pointer nor vector
 1863 | #define GET_LE_10BIT_LEFT_AS_8(_ptr) (_ptr[1] )
      |                                           ^
evg/stencil.c:2225:14: note: in expansion of macro ‘GET_LE_10BIT_LEFT_AS_8’
 2225 |         vv = GET_LE_10BIT_LEFT_AS_8(p_src + _this->off_v);
      |              ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [../config.mak:184: evg/stencil.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/marillat/gpac-dmo-2.2.0/src'
````

**Comment by aureliendavid** [comment]
I don't have a powerpc at hand to test it (I will try to set up one), but at first glance I'd say there are missing parenthesis here: 

https://github.com/gpac/gpac/blob/4669ba2293cf62fe1edf2ec2503729a0527c1f14/src/evg/stencil.c#L1863

you can try replacing the line with 

```c
#define GET_LE_10BIT_LEFT_AS_8(_ptr) ((_ptr)[1] )
```

and see if it builds

**Comment by aureliendavid** [comment]
> you can try replacing the line with
> 
> ```c
> #define GET_LE_10BIT_LEFT_AS_8(_ptr) ((_ptr)[1] )
> ```

or maybe even

```c
#define GET_LE_10BIT_LEFT_AS_8(_ptr) ( (u16 *)(_ptr)[1] )
```

**Comment by marillat** [comment]
Thanks, the first change fix this issue

```
--- a/src/evg/stencil.c
+++ b/src/evg/stencil.c
@@ -1860,7 +1860,7 @@ u32 get_pix_yuv420p(EVG_Texture *_this,
 #define GET_LE_10BIT_AS_16(_ptr) ( (((u16)(_ptr)[1])<<8 | (u16)(_ptr)[0] ) << 6 )
 
 //#define GET_LE_10BIT_LEFT_AS_8(_ptr) ( (((u16)(_ptr)[1])<<8 | (u16)(_ptr)[0] ) >> 8 )
-#define GET_LE_10BIT_LEFT_AS_8(_ptr) (_ptr[1] )
+#define GET_LE_10BIT_LEFT_AS_8(_ptr) ((_ptr)[1] )
 #define GET_LE_10BIT_LEFT_AS_16(_ptr) ( (((u16)(_ptr)[1])<<8 | (u16)(_ptr)[0] ) & 0xFFC0 )
 
 #define GET_BE_10BIT_AS_8(_ptr) ( (*(u16 *)(_ptr)) >> 2 )
```

**Comment by aureliendavid** [comment]
fix included in [b0ad643](https://github.com/gpac/gpac/commit/b0ad643014e567f47d1231771b473d25986c512e)
